compiler/back_end/cpp/testcode/enum_case_test.cc

```diff
@@ -0,0 +1,219 @@
+  EXPECT_TRUE(TryToGetEnumFromName("THREE_WORD_ENUM", &result));
+  EXPECT_EQ(EnumShoutyAndKCamel::THREE_WORD_ENUM, result);
+  EXPECT_TRUE(TryToGetEnumFromName("LONG_ENUM_VALUE_NAME", &result));
+  EXPECT_EQ(EnumShoutyAndKCamel::LONG_ENUM_VALUE_NAME, result);
+}
+
+TEST(EnumShoutyAndKCamel, EnumToName) {
+  EXPECT_EQ("FIRST", TryToGetNameFromEnum(EnumShoutyAndKCamel::FIRST));
+  EXPECT_EQ("SECOND", TryToGetNameFromEnum(EnumShoutyAndKCamel::SECOND));
+  EXPECT_EQ("TWO_WORD", TryToGetNameFromEnum(EnumShoutyAndKCamel::TWO_WORD));
+  EXPECT_EQ("THREE_WORD_ENUM",
+    TryToGetNameFromEnum(EnumShoutyAndKCamel::THREE_WORD_ENUM));
+  EXPECT_EQ("LONG_ENUM_VALUE_NAME",
+    TryToGetNameFromEnum(EnumShoutyAndKCamel::LONG_ENUM_VALUE_NAME));
+
+  EXPECT_EQ("FIRST", TryToGetNameFromEnum(EnumShoutyAndKCamel::kFirst));
+  EXPECT_EQ("SECOND", TryToGetNameFromEnum(EnumShoutyAndKCamel::kSecond));
+  EXPECT_EQ("TWO_WORD", TryToGetNameFromEnum(EnumShoutyAndKCamel::kTwoWord));
+  EXPECT_EQ("THREE_WORD_ENUM",
+    TryToGetNameFromEnum(EnumShoutyAndKCamel::kThreeWordEnum));
+  EXPECT_EQ("LONG_ENUM_VALUE_NAME",
+    TryToGetNameFromEnum(EnumShoutyAndKCamel::kLongEnumValueName));
+}
+
+TEST(EnumMixed, NameToEnum) {
+  EnumMixed result;
+
+  EXPECT_TRUE(TryToGetEnumFromName("FIRST", &result));
+  EXPECT_EQ(EnumMixed::FIRST, result);
+  EXPECT_TRUE(TryToGetEnumFromName("kFirst", &result));
+  EXPECT_EQ(EnumMixed::kFirst, result);
+  EXPECT_TRUE(TryToGetEnumFromName("SECOND", &result));
+  EXPECT_EQ(EnumMixed::SECOND, result);
+  EXPECT_TRUE(TryToGetEnumFromName("kTwoWord", &result));
+  EXPECT_EQ(EnumMixed::kTwoWord, result);
+  EXPECT_TRUE(TryToGetEnumFromName("THREE_WORD_ENUM", &result));
+  EXPECT_EQ(EnumMixed::THREE_WORD_ENUM, result);
+  EXPECT_TRUE(TryToGetEnumFromName("kThreeWordEnum", &result));
+  EXPECT_EQ(EnumMixed::kThreeWordEnum, result);
+  EXPECT_TRUE(TryToGetEnumFromName("kLongEnumValueName", &result));
+  EXPECT_EQ(EnumMixed::kLongEnumValueName, result);
+ }
+
+TEST(EnumMixed, NameToEnumFailsWithWrongCases) {
+  EnumMixed result = EnumMixed::FIRST;
+  EXPECT_FALSE(TryToGetEnumFromName("kSecond", &result));
+  EXPECT_EQ(EnumMixed::FIRST, result);
+  EXPECT_FALSE(TryToGetEnumFromName("TWO_WORD", &result));
+  EXPECT_EQ(EnumMixed::FIRST, result);
+  EXPECT_FALSE(TryToGetEnumFromName("LONG_ENUM_VALUE_NAME", &result));
+  EXPECT_EQ(EnumMixed::FIRST, result);
+}
+
+TEST(EnumMixed, EnumToName) {
+  EXPECT_EQ("FIRST", TryToGetNameFromEnum(EnumMixed::FIRST));
+  EXPECT_EQ("FIRST", TryToGetNameFromEnum(EnumMixed::kFirst));
+  EXPECT_EQ("SECOND", TryToGetNameFromEnum(EnumMixed::SECOND));
+  EXPECT_EQ("kTwoWord", TryToGetNameFromEnum(EnumMixed::kTwoWord));
+  EXPECT_EQ("kThreeWordEnum",
+    TryToGetNameFromEnum(EnumMixed::THREE_WORD_ENUM));
+  EXPECT_EQ("kThreeWordEnum",
+    TryToGetNameFromEnum(EnumMixed::kThreeWordEnum));
+  EXPECT_EQ("kLongEnumValueName",
+    TryToGetNameFromEnum(EnumMixed::kLongEnumValueName));
+}
+
```

Comment:
Please add test cases for the Emboss text format, using `WriteToString()` and `UpdateFromText()`.

The text format in general should not be affected by `[(cpp) enum_case]`, although I would be fine with having an option to use/allow (respectively) the C++ names instead of the Emboss names.

Reply:
Ah, good catch! Added test cases for these. I agree, the text format should not change, which it incorrectly did before fixing in the latest commit. Just like with enum from string/string from enum, it could be useful to optionally allow this, but that can be added if and when there is a use case.